File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/secret/impl/InMemorySecretRepository.java
package at.tugraz.tc.cyfile.secret.impl;

import at.tugraz.tc.cyfile.secret.Secret;
import at.tugraz.tc.cyfile.secret.SecretRepository;

public class InMemorySecretRepository implements SecretRepository {

    private Secret secret;

    public InMemorySecretRepository() {
    }

    public InMemorySecretRepository(Secret secret) {
        this.secret = secret;
    }

    @Override
    public Secret getSecret() {
        return secret;
    }

    @Override
    public boolean saveSecret(Secret secret) {
        if (secret == null) {
            return false;
        } else {
            this.secret = secret;
            return true;
        }
    }
}

File: CyFile-Android/app/src/main/kotlin/at/tugraz/tc/cyfile/secret/impl/InMemorySecretRepository.kt
package at.tugraz.tc.cyfile.secret.impl

import at.tugraz.tc.cyfile.secret.Secret
import at.tugraz.tc.cyfile.secret.SecretRepository

class InMemorySecretRepository : SecretRepository {

    private var secret: Secret? = null

    override fun retrieveSecret(): Secret? {
        return secret
    }

    override fun saveSecret(secret: Secret): Boolean {
        return if (this.secret == null) {
            this.secret = secret
            true
        } else {
            false
        }

    }


}
